q2_diversity/_beta/_visualizer.py

```diff
@@ -365,6 +365,12 @@ def adonis(output_dir: str,
         for i in t.factors:
             metadata.get_column(i.name())
 
+    # Check metadata for NaNs
+    metadata = metadata.to_dataframe()
+    if metadata.isnull().values.any() == True:
+        raise ValueError("Adonis cannot run with NaNs in metadata.")
+    metadata = qiime2.Metadata(metadata)
```

Comment:
```suggestion
    metadata_df = metadata.to_dataframe()
    if metadata_df.isnull().values.any():
        raise ValueError("Adonis cannot run with NaNs in metadata.")
```

Reply:
Yes, your suggested changes look great! I noticed the adonis test file, but was not sure if it was necessary/what would need to be done for that. Can you please explain what needs to be done to create the test?

Reply:
Thanks @collin-le!

I think in the adonis test suite we could just copy a test like this:

https://github.com/qiime2/q2-diversity/blob/406fedf67a574ecd8885f63dc58e06bd3ff5147a/q2_diversity/tests/test_adonis.py#L78-L84

and change the test data to be something with NaNs in the metadata. Then, the `self.assertRaisesRegex` part could match on some part of your error message above (`"Adonis cannot run with NaNs in metadata."`).